packages/functional-tests/src/tests/sound-test.ts

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as MRESDK from '@microsoft/mixed-reality-extension-sdk';
+import { Actor } from '@microsoft/mixed-reality-extension-sdk';
+import App from '../app';
+import delay from '../utils/delay';
+import destroyActors from '../utils/destroyActors';
+import Test from './test';
+
+export default class SoundTest extends Test {
+
+    constructor(app: App, private baseUrl: string) {
+        super(app);
+    }
+
+    public async run(): Promise<boolean> {
+        let success = true;
+
+        success = success && await this.runSoundTest();
+
+        return success;
+    }
+
+    public async runSoundTest(): Promise<boolean> {
+        // Make a root object.
+        const tester = MRESDK.Actor.CreateEmpty(this.app.context, {});
+
+        const textPromise = Actor.CreateEmpty(this.app.context, {
+            actor: {
+                name: 'label',
+                parentId: tester.value.id,
+                transform: {
+                    position: { x: 0, y: 2, z: 0 }
+                },
+                text: {
+                    anchor: MRESDK.TextAnchorLocation.MiddleCenter,
+                    color: { r: 30 / 255, g: 206 / 255, b: 213 / 255 },
+                    height: 0.3
+                }
+            }
+        });
+
+        const text = textPromise.value;
+        await textPromise;
+        text.text.contents = "Preloading Sound";
+
+        const soundId = this.app.context.assetManager.loadSound(
+            'sounds', `${this.baseUrl}/testsound.wav`,
+            {
+                looping: true
+            });
+
+        await soundId;
+        await delay(1 * 1000);
+
+        text.text.contents = "Starting Sound";
+        const instanceHandle = text.playSound(await soundId,
```

Comment:
>instanceHandle [](start = 14, length = 14)

I think an interface wrapper would be a more appropriate design here. Something like `SoundInstance`.
```ts
class SoundInstance {
    public setSoundState(state: Partial<SoundState>);
    public play();
    public pause();
    ...etc
};
```

Reply:
I'm thinking the same may apply to animations. I don't love that the animation API is on Actor.

---
In reply to: [256952297](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/191#discussion_r256952297) [](ancestors = 256952297)

Reply:
I agree this is better for soundinstance. Don't know if it's better for animations - unlike sound, there's not really animation instance handles created/destroyed at start/stop. And animation names are not synced from client to server, right? I guess there are workarounds for that, though. Regardless, this PR is for sound, so and I'll follow this pattern.

---
In reply to: [256953209](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/191#discussion_r256953209) [](ancestors = 256953209,256952297)